js/src/forum/states/DiscussionListState.js

```diff
@@ -77,17 +77,23 @@ export default class DiscussionListState {
   }
 
   /**
-   * Clear and reload the discussion list.
+   * Clear and reload the discussion list. Passing the option `deferClear: true`
+   * will clear discussions only after new data has been received.
+   * This can be used to refresh discussions without loading animations.
    */
-  refresh({ clear = true } = {}) {
+  refresh({ deferClear = false } = {}) {
     this.loading = true;
 
-    if (clear) {
+    if (!deferClear) {
       this.clear();
     }
 
     return this.loadResults().then(
       (results) => {
+        // Same effect as this.clear(), but without an additional redraw.
+        // This ensures that any changes made while waiting on this request
+        // are ignored. Otherwise, we could get duplicate discussions.
```

Comment:
I'm unsure about that "duplication" statement. This comment is inside of the promise callback, after everything has been done. There's no waiting at this point, no?

The reason to skip a redraw is just for performance and to maybe prevent a flicker?

Reply:
Well, this is the part that actually fixes the issue where changes to the discussion list while the refresh request is loading result in duplicate things.

It's explaining why we clear it, not why we clear it without `this.clear`. I'll reword this.